src/components/DatePicker/index.android.js

```diff
@@ -8,87 +8,76 @@ import CONST from '../../CONST';
+    const dateAsText = value || defaultValue ? moment(value || defaultValue).format(CONST.DATE.MOMENT_FORMAT_STRING) : '';
 
-        return (
-            <>
-                <TextInput
-                    label={this.props.label}
-                    accessibilityLabel={this.props.label}
-                    accessibilityRole={CONST.ACCESSIBILITY_ROLE.TEXT}
-                    value={dateAsText}
-                    forceActiveLabel
-                    placeholder={this.props.placeholder}
-                    errorText={this.props.errorText}
-                    containerStyles={this.props.containerStyles}
-                    textInputContainerStyles={this.state.isPickerVisible ? [styles.borderColorFocus] : []}
-                    onPress={this.showPicker}
-                    editable={false}
-                    disabled={this.props.disabled}
-                    onBlur={this.props.onBlur}
-                    ref={(el) => {
-                        if (!_.isFunction(this.props.innerRef)) {
-                            return;
-                        }
-                        if (el && el.focus && typeof el.focus === 'function') {
-                            let inputRef = {...el};
-                            inputRef = {...inputRef, focus: this.showPicker};
-                            this.props.innerRef(inputRef);
-                            return;
-                        }
+    return (
+        <>
+            <TextInput
+                label={label}
+                accessibilityLabel={label}
+                accessibilityRole={CONST.ACCESSIBILITY_ROLE.TEXT}
+                value={dateAsText}
+                forceActiveLabel
+                placeholder={placeholder}
+                errorText={errorText}
+                containerStyles={containerStyles}
+                textInputContainerStyles={isPickerVisible ? [styles.borderColorFocus] : []}
+                onPress={() => showPicker()}
+                editable={false}
+                disabled={disabled}
+                onBlur={onBlur}
+                ref={(el) => {
+                    if (!_.isFunction(innerRef)) {
+                        return;
+                    }
+                    if (el && el.focus && typeof el.focus === 'function') {
+                        let inputRef = {...el};
+                        inputRef = {...inputRef, focus: showPicker};
+                        innerRef(inputRef);
+                        return;
+                    }
 
-                        this.props.innerRef(el);
-                    }}
+                    innerRef(el);
+                }}
+            />
+            {isPickerVisible && (
+                <RNDatePicker
+                    value={value || defaultValue ? moment(value || defaultValue).toDate() : new Date()}
+                    mode="date"
+                    onChange={() => setDate()}
```

Comment:
```suggestion
                    onChange={setDate}
```

It could be simplified.

Reply:
I agree that that'd be better/simpler. The reason I did it that way is because of our linter.
![image](https://github.com/Expensify/App/assets/55854142/72afe631-1d31-437d-a0c8-e6f9e152a543)

Reply:
Oh.. ok 🤷‍♂️